### ISSUE TITLE
Graph does not respect wildcard exclusions

### ISSUE THREAD

**Issue by crosslandwa** [ferstl/depgraph-maven-plugin#76]
Hi - great tool, I've been finding it really useful understanding dependencies in some legacy apps I've been working on.

I've identified an issue where the generated dependency graph incorrectly shows dependencies that maven has excluded via a wildcard.

If I have an `app`, which depends on `lib-b`, which depends on `lib-c`:

#### app pom:
```
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.github.crosslandwa</groupId>
    <artifactId>mvndependencygraph-issue-app</artifactId>
    <packaging>jar</packaging>
    <version>1.0</version>
    <dependencies>
        <dependency>
            <groupId>com.github.crosslandwa</groupId>
            <artifactId>mvndependencygraph-issue-lib-b</artifactId>
            <version>1.0</version>
            <exclusions>
                <exclusion>
                    <groupId>com.github.crosslandwa</groupId>
                    <artifactId>*</artifactId>
                </exclusion>
            </exclusions>
        </dependency>
    </dependencies>
</project>
```

#### lib-b pom:
```
<project xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xmlns="http://maven.apache.org/POM/4.0.0"
         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
    <modelVersion>4.0.0</modelVersion>
    <groupId>com.github.crosslandwa</groupId>
    <artifactId>mvndependencygraph-issue-lib-b</artifactId>
    <packaging>jar</packaging>
    <version>1.0</version>
    <dependencies>
        <dependency>
            <groupId>com.github.crosslandwa</groupId>
            <artifactId>mvndependencygraph-issue-lib-c</artifactId>
            <version>1.0</version>
        </dependency>
    </dependencies>
</project>
```

And I generate a graph via
```
mvn com.github.ferstl:depgraph-maven-plugin:graph -DcreateImage=true -DshowGroupIds=true -DshowVersions=true -DshowConflicts=true -DshowDuplicates=true
```

#### Expected:
The output for `mvn dependency:tree`
```
[INFO] --------< com.github.crosslandwa:mvndependencygraph-issue-app >---------
[INFO] Building mvndependencygraph-issue-app 1.0
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ mvndependencygraph-issue-app ---
[INFO] com.github.crosslandwa:mvndependencygraph-issue-app:jar:1.0
[INFO] \- com.github.crosslandwa:mvndependencygraph-issue-lib-b:jar:1.0:compile
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
```

Which would have the expected graph of:

![dependency-graph](https://user-images.githubusercontent.com/4037089/42943252-2047965a-8b5a-11e8-99df-cf2d4e5831f2.png)

#### Actual:
But I'm actually seeing...

![dependency-graph](https://user-images.githubusercontent.com/4037089/42943313-4ac1f25e-8b5a-11e8-824f-9d0e95f81c3c.png)

_**Note 1**: explicitly defining an exclusion with a groupID works as expected, e.g._
```
<dependency>
    <groupId>com.github.crosslandwa</groupId>
    <artifactId>mvndependencygraph-issue-lib-b</artifactId>
    <version>1.0</version>
    <exclusions>
        <exclusion>
            <groupId>com.github.crosslandwa</groupId>
            <artifactId>mvndependencygraph-issue-lib-c</artifactId>
        </exclusion>
    </exclusions>
</dependency>
```

_**Note 2**: ommiting **both** the `-DshowConflicts=true` and  `-DshowDuplicates=true` options also works correctly, i.e._
```
mvn com.github.ferstl:depgraph-maven-plugin:graph -DcreateImage=true -DshowGroupIds=true -DshowVersions=true
```

**Comment by ferstl** [comment]
Hi

Thank you for the positive feedback and your detailed description!
The reason for your problem is that as soon as you enable `showConflicts` or `showDuplicates` the plugin will use the old Maven-2 dependency resolution mechanism, which *may produce not the same result* and does not support dependency resolution features (such as wildcard exclusions) that were introduced in newer Maven versions. I mentioned this in the [plugin documentation](https://ferstl.github.io/depgraph-maven-plugin/graph-mojo.html#showDuplicates) (not very visibly, I have to admit).

Internally, the plugin uses an old version of the [maven-dependency-tree](http://search.maven.org/#artifactdetails%7Corg.apache.maven.shared%7Cmaven-dependency-tree%7C2.2%7Cjar) library, which creates the object model for the dependency graph. The feature for reporting duplicates and conflicts was eliminated in newer versions.
To correctly report duplicates and conflicts I would need to replace the maven-dependency-tree library with my own implementation, which is quite some work.

So for now, I would suggest not to use `showConflicts` or `showDuplicates` when you need the wildcard exclusions.

I have been thinking of removing the `showConflicts` or `showDuplicates` options from the plugin but I find them quite useful even though the graph might not be accurate in some edge cases.

**Comment by ferstl** [comment]
The fix for #7 does also fix this problem.

**Comment by ferstl** [comment]
Released version 3.2.0 of the plugin
